### PR TITLE
refactor nodata

### PR DESCRIPTION
**Pull request by pangthebabydustbunny**

### CODE REVIEW

**Comment by evilpeach** [comment]
Please fix the conflict.

**Comment by pangthebabydustbunny** [comment]
> Please fix the conflict.

fixed ka

File: src/components/LatestRequests.re [hunk 1]
```diff
@@ -183,32 +183,26 @@ let make = (~latest5RequestSub: Sub.t(array(RequestSub.t))) => {
            </Row>
          </THead>}
     {switch (latest5RequestSub) {
+     | Data(requests) when requests->Belt.Array.length === 0 =>
+       <EmptyContainer height={`calc((`sub, `percent(100.), `px(86)))} boxShadow=true>
+         <img
+           src={isDarkMode ? Images.noDataDark : Images.noDataLight}
```
Comment (evilpeach) [inline src/components/LatestRequests.re:189]: alt?

File: src/pages/DataSourceHomePage.re [hunk 2]
```diff
@@ -267,40 +267,39 @@ let make = () => {
                  </Row>
                </THead>}
           {switch (allSub) {
+           | Data((dataSources, _)) when dataSources->Belt.Array.length === 0 =>
+             <EmptyContainer>
+               <img
+                 src={isDarkMode ? Images.noOracleDark : Images.noOracleLight}
```
Comment (evilpeach) [inline src/pages/DataSourceHomePage.re:273]: alt?

File: src/pages/OracleScriptHomePage.re [hunk 3]
```diff
@@ -472,50 +471,49 @@ let make = () => {
                  </Row>
                </THead>}
           {switch (allSub) {
+           | Data((oracleScripts, _)) when oracleScripts->Belt.Array.length === 0 =>
+             <EmptyContainer>
+               <img
+                 src={isDarkMode ? Images.noOracleDark : Images.noOracleLight}
```
Comment (evilpeach) [inline src/pages/OracleScriptHomePage.re:477]: alt?

File: src/utils/Format.re [hunk 4]
```diff
@@ -7,7 +7,7 @@ let withCommas = value =>
          |> Js.String.split("")
          |> Js.Array.reverseInPlace
          |> Js.Array.reducei(
-              (acc, ch, idx) => idx mod 3 == 0 && idx != 0 ? ch ++ "," ++ acc : ch ++ acc,
+              (acc, ch, idx) => idx mod 3 == 0 && idx !== 0 ? ch ++ "," ++ acc : ch ++ acc,
```
Comment (evilpeach) [inline src/utils/Format.re:10]: mod 3 === 0

File: src/reusable/Select.re [hunk 5]
```diff
@@ -41,13 +41,15 @@ let make = (~options, ~setSelectOption, ~selectLabel, ~selectedOption, ~disabled
...
-           : React.null}
+        {switch (options |> Belt.Array.length) {
+         | 0 =>
+           options
+           |> Belt.Array.map(_, value =>
+                <option key=value value> {value |> React.string} </option>
+              )
+           |> React.array
+         | _ => React.null
+         }}
```
Comment (evilpeach) [inline src/reusable/Select.re:52]: I think the old is ok.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by pzshine — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**